### DISCUSSION TITLE
Making connections with like minded folks: LabOP<->LinkML

### DISCUSSION THREAD

**Discussion by photocyte** [Bioprotocols/labop#212] [General]
Hi all, 

I spoke a bit to Jonny (@sneakers-the-rat) about making connections between [LabOP](https://github.com/Bioprotocols/labop) & [LinkML](https://github.com/linkml/linkml), since there seems to be some potential synergy there (sorry it took me so long to start this connection! Hopefully it is a good time). 

@danbryce , @jakebeal , @bbartley  maybe you could take a look at the technical side of [LinkML](https://github.com/linkml/linkml) and ponder whether it might be a good fit for any areas of [LabOP](https://github.com/Bioprotocols/labop)?

edit: CC @markdoerr as well, to add to the group of folks who have a much deeper intuition for `RDF` machinery than I do!

CCing @cmungall as well, since he is the biggest LOC contributor to LinkML and coming from the biology ontology side might appreciate LabOP more than the average bear...

Edit: linking to the LinkML Mastodon account as that has some nice recent summaries of progress: https://fosstodon.org/@linkml

All the best,
-Tim (photocyte , UCSD/SIO)

**Comment by sneakers-the-rat** [comment]
what's the source of the labOP ontology/schema/etc.? I find this https://github.com/Bioprotocols/labop/blob/develop/labop/container-ontology.ttl but readme says it's a serialization of the ontology, what does that get built from?

**Reply by rpgoldman** [reply]
I developed it using Protege.  The turtle format is one of a number of alternative output formats. It's not an export of a subset of the model, it is a full representation of the model.

**Reply by photocyte** [reply]
I might emphasize that `.ttl` is specifically the `container-ontology` , which is an extension of sorts of LabOP. Other relevant `.ttl` might be found in this dir: https://github.com/Bioprotocols/labop/tree/main/labop/lib .

**Reply by rpgoldman** [reply]
The container-ontology is a stand-alone ontology, actually, and is employed (sometimes) by LabOP.  If I recall correctly, its concepts are all defined independently of LabOP proper (to avoid circularities).

**Comment by sneakers-the-rat** [comment]
Before i participate, is the listing of Raytheon BBN as one of the copyright holders current and correct? 
https://github.com/Bioprotocols/labop/blob/2e2bd88150c71a440771fd369f40295dfd622324/LICENSE.txt#L1

**Reply by photocyte** [reply]
Here is past issue discussion on it: https://github.com/Bioprotocols/labop/issues/71 , I checked the Bioprotocols Working Group documents (https://drive.google.com/drive/search?q=license%20parent:1JBHfEfrxVF4VdH9jKIXCKUkE7a1H8484) and there wasn't recorded verbal discussion on it - I believe the PR + license happened before the Bioprotocols community and regular meetings + minutes got started. There is now an Opencollective that represents the community and its funding explicitly: https://opencollective.com/bioprotocols#category-ABOUT , but it has been pretty quiet. I'm assuming you propose the copyright be transferred to the OpenCollective? I'm not sure how to go about that, and, in practical terms the `MIT` license is pretty permissive. Is the concern that it is potentially revokable?

**Reply by sneakers-the-rat** [reply]
I wish no offense to those employed there, but I don't want to contribute to anything from Raytheon or its subsidiaries. Hope you understand, nothing personal, just principle.

**Reply by photocyte** [reply]
Got it! Will reach out again if that attribution can be transferred. Nontheless I hope you agree that the project is being run in an open way that is in line with open source and copyleft principles (but of course appreciate any tips where it isn't). The project was initially funded from DARPA funds - not sure if that also precludes. That provenance I cannot change.

**Reply by sneakers-the-rat** [reply]
again no offense intended, but taking a look at the example notebook it seems like we are probably pretty far apart on what we are identifying as the primary barriers/priorities for scientific/data tooling here. there are also some technical/design differences that I think would make it pretty hard for me to be a collaborator here (eg. 110 seconds to import the package is a lot). wish y'all best of luck, but I don't think I can be of much use here.

**Reply by photocyte** [reply]
That's helpful feedback Jonny, I appreciate your taking a look! If any parts in the quick look, looked like LinkML might be applicable, that'd be great to know. I'm still wrapping my head around LinkML - that's on me as my expertise in software dev & linked data stuff is limited, rather than anything LinkML is doing as it's documentation looks great

**Comment by cmungall** [comment]
Hi all! I'm very interested in both sbol/synbio and lab automation schemas. I haven't taken a close look at labop yet, would one of the aims be to have a way of precisely specifying a protocol such that it could be handled by a robot, e.g. like ECL https://www.emeraldcloudlab.com/how-it-works/command/

It looks like labop is layered on sbol, which makes sense. Would it also make sense to first think about what a linkml-ization of sbol would look like?

**Reply by photocyte** [reply]
Hi there Chris, yes, LabOP has done a pilot project with ECL, and LabOP is also inspired by Strateos's analogous Autoprotocol format (but aims to be more extensible). So robotic automation is plainly within the scope, but bench scienctist protocols are also in scope. 

I'm not super familiar with SBOL so will let others chime in there!
